### PR TITLE
Add Unit Tests for CreateOfferingsComponent

### PR DESCRIPTION
**Pull request by milicaradicc**
This PR adds comprehensive unit tests for the CreateOfferingsComponent, significantly improving test coverage and ensuring the reliability of the service creation functionality.
Changes Made
## Test Coverage Improvements

Fixed Angular Material imports: Resolved NG01203 error by properly configuring Material modules and removing duplicate imports
Added comprehensive form validation tests: Coverage for required fields, price/discount validation, and form state management
Implemented service interaction tests: Mocking and testing ServiceService, AuthService, and CategoryService calls
Added error handling tests: Testing both success and failure scenarios with appropriate user feedback

## Test Configuration Fixes

Properly organized Angular Material module imports
Added BrowserAnimationsModule for Material components compatibility
Removed duplicate module imports that were causing conflicts
Enhanced mock service setup with realistic data structures

## New Test Cases

Component Initialization: Verifies component creation and initial state
Form Validation: Tests required field validation and form state changes
Category Management: Tests both existing category selection and new category creation flows
Service Submission: Validates proper DTO construction and service calls
Time Type Handling: Tests both fixed and flexible duration options
User Feedback: Verifies success/error message display
Form Reset: Ensures form is properly reset after successful submission
Edge Cases: Price/discount validation, boundary conditions

Quality Assurance
## Test Results

All 14 test cases passing
No Angular Material import conflicts
Proper form control value accessor resolution
Comprehensive coverage of component functionality

### CODE REVIEW

File: EventPlanner/src/app/offering/create-offerings/create-offerings.component.spec.ts [hunk 1]
```diff
@@ -1,23 +1,456 @@
...
+
+    expect(mockServiceService.add).toHaveBeenCalledWith(
+      jasmine.objectContaining({
+        pending: true
+      })
+    );
+  });
+
+  it('should validate required fields correctly', () => {
+    // Test praznih obaveznih polja
```
Comment (marina-ivanovic) [inline EventPlanner/src/app/offering/create-offerings/create-offerings.component.spec.ts]: Remove this comment or translate it, some other comments in this file could also be removed but you don't have to

**Review by marina-ivanovic — CHANGES_REQUESTED** [review]
Just a small comment removal needed, other than that, the tests are well written and work

**Review by marina-ivanovic — APPROVED** [review]
The tests are clean and well thought out, well done
